src/app/impl_self.rs

```diff
@@ -33,8 +33,27 @@ impl App {
         };
     }
 
+    pub fn list_habit(&self) -> Vec<String> {
+        let mut habits_names: Vec<String> = vec![];
```

Comment:
I would suggest trimming this down to not require a mutable vector:
```suggestion
        let habits_names = self.habits.iter().map(|x| x.name()).collect::<Vec<_>>();
```